inspirehep/modules/migrator/tasks.py

```diff
@@ -154,6 +159,52 @@ def migrate_chunk(chunk, broken_output=None, dry_run=False):
         models_committed.connect(record_modification)
 
 
+@celery.task(ignore_result=True)
+def add_citation_counts():
+    from elasticsearch.helpers import bulk as es_bulk
+    from elasticsearch.helpers import scan as es_scan
+    from collections import Counter
+
+    # list of update-jsons for each record
+    records_to_update = []
+
+    # lookup dictionary where key: recid of the record
+    # and value: number of records that cite that record
+    citations_lookup = Counter()
+    for record in es_scan(
+            es,
+            query={
+                "query": {
+                    "match_all": {}},
```

Comment:
Also you can search for only those records that have actually anything in `references.recid`.